includes/diagnostics/class-wc-stripe-diagnostics-recorder.php

```
@@ -0,0 +1,400 @@
+				'order_id'   => $order_id,
+				'session_id' => $session_id,
+			]
+		);
+		$this->store->append_event( $session_id, $redacted );
+		$this->promoter->maybe_promote( $session_id, $redacted );
+
+		// Webhook-only traces also spend a slot from the capture budget.
+		if ( class_exists( 'WC_REST_Stripe_Diagnostics_Controller' ) ) {
+			WC_REST_Stripe_Diagnostics_Controller::enforce_capture_limit( $this->store );
+		}
+	}
```

PHPMD (2.15.0)

[warning] The method on_webhook_received() has a Cyclomatic Complexity of 13. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)